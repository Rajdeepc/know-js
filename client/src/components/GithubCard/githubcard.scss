.githubCard {
    height: 200px;
    margin-bottom: 20px;
    .card-title {
        display: block;
        display: -webkit-box;
        -webkit-line-clamp: 2;
        -moz-line-clamp: 2;
        -ms-line-clamp: 2;
        -o-line-clamp: 2;
        line-clamp: 2;
        -webkit-box-orient: vertical;
        -moz-box-orient: vertical;
        -ms-box-orient: vertical;
        -o-box-orient: vertical;
        box-orient: vertical;
        overflow: hidden;
        text-overflow: ellipsis;
        white-space: normal;
        font-weight: 700;
        height: 42px;
        font-size: 15px;
        margin-bottom: 10px;
    }

    p.card-text {
        display: block;
        display: -webkit-box;
        -webkit-line-clamp: 3;
        -moz-line-clamp: 3;
        -ms-line-clamp: 3;
        -o-line-clamp: 3;
        line-clamp: 3;
        -webkit-box-orient: vertical;
        -moz-box-orient: vertical;
        -ms-box-orient: vertical;
        -o-box-orient: vertical;
        box-orient: vertical;
        overflow: hidden;
        text-overflow: ellipsis;
        white-space: normal;
        height: 56px;
        font-size: 12px;
        margin-bottom: 10px;
    }

    a.card-link {
        font-size: 14px;
        color: #7d7d7d;
        svg{
            margin-right: 4px;
            vertical-align: sub;
        }
    }
}