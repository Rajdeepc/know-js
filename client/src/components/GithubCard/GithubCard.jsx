import React from "react";
import { Row,Col,Card } from "react-bootstrap";
import "./githubcard.scss";
import { RiStarSLine } from "react-icons/ri";
import { GoIssueOpened } from "react-icons/go";
import PropTypes from 'prop-types';

const GithubCard = ({ item }) => {

  return (
    <Card className="githubCard" data-testid="test-github-card-wrapper">
      <Card.Body>
        <a href="#" rel="noopener noreferrer" target="_blank" href={item.html_url}>
        <Card.Title>{item.full_name}</Card.Title>
        <Card.Subtitle className="mb-2 text-muted">
          <Card.Text>{item.owner.login}</Card.Text>
        </Card.Subtitle>
        <Card.Text className="desc">{item.description}</Card.Text>
        <Row>
          <Col>
          <Card.Text>
          <RiStarSLine size={20}/>
          {item.stargazers_count}
        </Card.Text>
          </Col>
          <Col className="text-right">
          <Card.Text>
          <GoIssueOpened size={20}/>
          {item.open_issues_count}
        </Card.Text>
          </Col>
        </Row>
        
        
        </a>
      </Card.Body>
    </Card>
  );
};


GithubCard.defaultProps = {
  html_url:'',
  full_name:'',
  owner: {
  },
  description:'',
  stargazers_count:0,
  open_issues_count:0
}

GithubCard.propTypes = {
  html_url:PropTypes.string,
  full_name: PropTypes.string,
  owner: PropTypes.objectOf(PropTypes.string),
  description:PropTypes.string,
  stargazers_count:PropTypes.number,
  open_issues_count: PropTypes.number
}

export default GithubCard;
