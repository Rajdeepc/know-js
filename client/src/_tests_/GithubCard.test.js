import React from "react";
import { render, cleanup } from "@testing-library/react";
import GithubCard from "../components/GithubCard/GithubCard";
import "@testing-library/jest-dom/extend-expect";

afterEach(cleanup);

const props = {
    item :{
        html_url: "",
        full_name: "",
        owner: {
          login: "",
        },
        description: "",
        stargazers_count: 0,
        open_issues_count: 0,
    }
  
};

describe("<GithubCard component>", () => {
  test("should be defined", () => {
    expect(GithubCard).toBeDefined();
  });
  test("renders GithubCard Component", () => {
    const { container, rerender } = render(<GithubCard {...props} />);
    rerender(<GithubCard {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });
  test("renders for default Props", () => {
      const { getByTestId } = render(
          <GithubCard {...props}
          />
      );
      expect(getByTestId('test-github-card-wrapper')).toBeDefined();
      expect(GithubCard.defaultProps.open_issues_count).toBeDefined()
  });
  //open_issues returns correctly
  test('open_issues returns correctly' , () => {
      const result = GithubCard.defaultProps.open_issues_count;
      expect(result).toBe(0)
  })
});
