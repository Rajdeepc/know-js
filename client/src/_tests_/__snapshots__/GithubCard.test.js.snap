// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GithubCard component> renders GithubCard Component 1`] = `
<div
  class="githubCard card"
  data-testid="test-github-card-wrapper"
>
  <div
    class="card-body"
  >
    <a
      href=""
      rel="noopener noreferrer"
      target="_blank"
    >
      <div
        class="card-title h5"
      />
      <div
        class="mb-2 text-muted card-subtitle h6"
      >
        <p
          class="card-text"
        />
      </div>
      <p
        class="desc card-text"
      />
      <div
        class="row"
      >
        <div
          class="col"
        >
          <p
            class="card-text"
          >
            <svg
              fill="currentColor"
              height="20"
              size="20"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M12 17l-5.878 3.59 1.598-6.7-5.23-4.48 6.865-.55L12 2.5l2.645 6.36 6.866.55-5.231 4.48 1.598 6.7L12 17zm0-2.344l2.817 1.72-.766-3.21 2.507-2.147-3.29-.264L12 7.708l-1.268 3.047-3.29.264 2.507 2.147-.766 3.21L12 14.657z"
                />
              </g>
            </svg>
            0
          </p>
        </div>
        <div
          class="text-right col"
        >
          <p
            class="card-text"
          >
            <svg
              fill="currentColor"
              height="20"
              size="20"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 14 16"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M7 2.3c3.14 0 5.7 2.56 5.7 5.7s-2.56 5.7-5.7 5.7A5.71 5.71 0 0 1 1.3 8c0-3.14 2.56-5.7 5.7-5.7zM7 1C3.14 1 0 4.14 0 8s3.14 7 7 7 7-3.14 7-7-3.14-7-7-7zm1 3H6v5h2V4zm0 6H6v2h2v-2z"
                fill-rule="evenodd"
              />
            </svg>
            0
          </p>
        </div>
      </div>
    </a>
  </div>
</div>
`;
